agent/s3/factory/factory.go

```diff
@@ -56,43 +58,55 @@ func (*s3ClientCreator) NewS3ManagerClient(bucket, region string,
 		WithCredentials(
 			awscreds.NewStaticCredentials(creds.AccessKeyID, creds.SecretAccessKey,
 				creds.SessionToken)).WithRegion(region)
+	if config.IsFIPSEnabled() {
+		logger.Debug("FIPS mode detected, using FIPS-compliant S3 endpoint")
+		cfg.UseFIPSEndpoint = endpoints.FIPSEndpointStateEnabled
+	}
 	sess := session.Must(session.NewSession(cfg))
 	svc := s3.New(sess)
 	bucketRegion, err := getRegionFromBucket(svc, bucket)
 	if err != nil {
 		return nil, err
 	}
-	sessWithRegion := session.Must(session.NewSession(cfg.WithRegion(bucketRegion)))
+	cfg.WithRegion(bucketRegion)
```

Comment:
Why was this factored out on it's own line? I don't think this is actually setting the region anymore as it doesn't change the config in-place. You would need to do `cfg := cfg.NewRegion(bucketRegion)`

see https://docs.aws.amazon.com/sdk-for-go/api/aws/#Config.WithRegion